### ISSUE TITLE
Investigate regression tests for n3fit

### ISSUE THREAD

**Issue by scarlehoff** [NNPDF/nnpdf#723]
One of the tests for `n3fit` is a regression test of a small fit, sitting at `n3fit/tests/test_fit.py`. What is does it to run a small fit (1000 epoch) and then compare the chi2 up to a number of decimal places.

There is a problem though, the result is not 100% consistent from computer to computer. My personal guess is that this is due to numerical precision blowing up (1000 epochs of training, with a few convolutions each, a training with a moving learning rate that depends on previous steps, etc) as the numbers, even when they are different, are too close to each other to be a difference in the generation of random numbers.

This is not weird, just changing the order of operations can have an effect on the remainders due to numerical precision. But for float32 this should be on the 8th decimal place and not the second one.

Long story short: it would be nice to find a configuration robust enough that can withstand small changes to the code.

Possible solution:
- Running the regression test fit on float64 (the default is float32)
